### PR TITLE
[Feat][Flink-K8s-V2] Refactor the lifecycle control of Flink K8s application-mode jobs

### LINKED ISSUE #2882: [Feature] [Flink-K8s-V2] Refactor the lifecycle control of Flink application-mode jobs on Kubernetes
It is a subtask of #2879 which includes the following content:
- Adapt [KubernetesNativeApplicationClient.doSubmit](https://github.com/apache/incubator-streampark/blob/dev/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesNativeApplicationClient.scala#L38) to [FlinkOperator. deployApplicationJob](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/FlinkK8sOperator.scala#L31);
- Adapt [KubernetesNativeApplicationClient.doCancel](https://github.com/apache/incubator-streampark/blob/dev/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesNativeApplicationClient.scala#L91) to  [FlinkOperator.cancelJob，FlinkOperator.stopJob](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/FlinkK8sOperator.scala#L51) ;
- Adapt [KubernetesNativeApplicationClient.doTriggerSavepoint](https://github.com/apache/incubator-streampark/blob/dev/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesNativeApplicationClient.scala#L100) to [FlinkOperator.triggerJobSavepoint](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/FlinkK8sOperator.scala#L56);
- Adapt KubernetesNativeApplicationClient.shutdown  to [FlinkK8sOperator.delete](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/FlinkK8sOperator.scala#L41);
- Remove the step 4 to 8 in [FlinkK8sApplicationBuildPipeline](https://FlinkK8sApplicationBuildPipeline) ;
- Remove the remaining useless code.

Ref: [design doc](https://www.craft.me/s/LrTTzYHqSOLYqY)

### PR DESCRIPTION
**Pull request by Al-assad**
## What changes were proposed in this pull request

Issue Number: close  #2882

## Brief change log
- Adapt [KubernetesNativeApplicationClient.doSubmit](https://github.com/apache/incubator-streampark/blob/dev/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesNativeApplicationClient.scala#L38) to [FlinkOperator. deployApplicationJob](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/FlinkK8sOperator.scala#L31);
- Adapt [KubernetesNativeApplicationClient.doCancel](https://github.com/apache/incubator-streampark/blob/dev/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesNativeApplicationClient.scala#L91) to [FlinkOperator.cancelJob，FlinkOperator.stopJob](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/FlinkK8sOperator.scala#L51) ;
- Adapt [KubernetesNativeApplicationClient.doTriggerSavepoint](https://github.com/apache/incubator-streampark/blob/dev/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesNativeApplicationClient.scala#L100) to [FlinkOperator.triggerJobSavepoint](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/FlinkK8sOperator.scala#L56);
- Create a new BuildPipeline for Flink application mode job resource 

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): no

### CODE REVIEW

**Comment by Al-assad** [comment]
Hi @caicancai, This PR is a fix for the code in PR #2994, and also a complete implementation of issue #2882. It can serve as a reference for your upcoming implementations of #2883 and #2884.

**Comment by Al-assad** [comment]
@wolfboys Please take a look.
Although this is a large-scale change, but the switch configuration `streampark.flink-k8s.enable-v2`  can be used to switch between the developing flink-k8s-v2 code and the old flink-k8s code. 
In the case of the default `streampark.flink-k8s.enable-v2` config being `false`, StreamPark will still use the original flink-k8s code. Therefore this PR will not take effect on existing StreamPark in practice.

**Comment by caicancai** [comment]
> Hi @caicancai, This PR is a fix for the code in PR #2994, and also a complete implementation of issue #2882. It can serve as a reference for your upcoming implementations of #2883 and #2884.

Thank you

**Comment by caicancai** [comment]
> @wolfboys Please take a look. Although this is a large-scale change, but the switch configuration can be used to switch between the developing flink-k8s-v2 code and the old flink-k8s code. In the case of the default config being , StreamPark will still use the original flink-k8s code. Therefore this PR will not take effect on existing StreamPark in practice.`streampark.flink-k8s.enable-v2``streampark.flink-k8s.enable-v2``false`

This is a good idea

File: streampark-common/src/main/scala/org/apache/streampark/common/conf/K8sFlinkConfig.scala [hunk 1]
```diff
@@ -18,37 +18,54 @@
...
+    key = "streampark.flink-k8s.enable-v2",
+    defaultValue = false,
+    classType = classOf[Boolean],
+    description =
+      "Whether to enable the v2 version(base on flink-kubernetes-operator) of flink kubernetes operation"
+  )
+
+  //  ======= deprecated =======
+
+  @deprecated
```
Comment (wolfboys) [inline streampark-common/src/main/scala/org/apache/streampark/common/conf/K8sFlinkConfig.scala:36]: StreamPark consists of two parts: a development framework and a stream processing platform. The development framework part provides some APIs to users. The parameter InternalOption was originally designed for internal use within the platform, It is a non-public API. Therefore, these parameters can be deleted directly without causing any other impact on users

**Comment by wolfboys** [comment]
> @wolfboys Please take a look. Although this is a large-scale change, but the switch configuration `streampark.flink-k8s.enable-v2` can be used to switch between the developing flink-k8s-v2 code and the old flink-k8s code. In the case of the default `streampark.flink-k8s.enable-v2` config being `false`, StreamPark will still use the original flink-k8s code. Therefore this PR will not take effect on existing StreamPark in practice.

Do k8sv2 and the current k8s native approach support parallel usage, or does k8sv2 replace the implementation of k8s native, only keep k8sv2?

**Comment by caicancai** [comment]
@Al-assad I don't seem to see the response implementation of the cancellation api

**Comment by caicancai** [comment]
> > 请看一看。虽然这是一个大规模的变化，但交换机配置可以用来在正在开发的 flink-k8s-v2 代码和旧的 flink-k8s 代码之间切换。在默认配置的情况下，StreamPark 仍将使用原始的 flink-k8s 代码。因此，此 PR 在实践中不会对现有的 StreamPark 生效。`streampark.flink-k8s.enable-v2``streampark.flink-k8s.enable-v2``false`
> 
> k8sv2 和当前的 k8s 原生方法是否支持并行使用，还是 k8sv2 取代了 k8s 原生的实现，只保留 k8sv2？

I think v2 is still in the development stage and is not stable yet. I will delete the v1 code after the development is tested and there are no problems.

**Comment by wolfboys** [comment]
> > > 请看一看。虽然这是一个大规模的变化，但交换机配置可以用来在正在开发的 flink-k8s-v2 代码和旧的 flink-k8s 代码之间切换。在默认配置的情况下，StreamPark 仍将使用原始的 flink-k8s 代码。因此，此 PR 在实践中不会对现有的 StreamPark 生效。`` streampark.flink-k8s.enable-v2``streampark.flink-k8s.enable-v2``false ``
> > 
> > 
> > k8sv2 和当前的 k8s 原生方法是否支持并行使用，还是 k8sv2 取代了 k8s 原生的实现，只保留 k8sv2？
> 
> I think v2 is still in the development stage and is not stable yet. I will delete the v1 code after the development is tested and there are no problems.

okay 👌 This is a relatively secure/reliable approach

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/application/impl/ApplicationActionServiceImpl.java [hunk 2]
```diff
@@ -413,10 +414,13 @@ public void start(Application appParam, boolean auto) throws Exception {
       extraParameter.put(ConfigConst.KEY_FLINK_SQL(null), flinkSql.getSql());
     }
 
+    // TODO Need to display more K8s submission parameters in the front-end UI.
+    //      See: org.apache.streampark.flink.client.bean.KubernetesSubmitParam
     KubernetesSubmitParam kubernetesSubmitParam =
-        new KubernetesSubmitParam(
+        KubernetesSubmitParam.apply(
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/application/impl/ApplicationActionServiceImpl.java:420]: good job

**Review by wolfboys — APPROVED** [review]
LGTM.  
There may still be some small issues, but let's merge this PR first and continue improving it later. This also makes it easier for other developers to collaborate and improve together.

**Pull request merged**